diff --git a/ament_package/templates.py b/ament_package/templates.py
index 8fda08d..56a0811 100644
--- a/ament_package/templates.py
+++ b/ament_package/templates.py
@@ -15,17 +15,33 @@
 import os
 import re
 
-try:
-    import importlib.resources as importlib_resources
-except ModuleNotFoundError:
-    import importlib_resources
+# try:
+#     import importlib.resources as importlib_resources
+# except ModuleNotFoundError:
+#     import importlib_resources
+
+import importlib.resources
+from importlib.resources import as_file, files
+# import importlib_resources
+# from importlib_resources import as_file, files
+
+# from .template import environment_hook, package_level, prefix_level
 
 IS_WINDOWS = os.name == 'nt'
 
+# importlib
+# DeprecationWarning: path is deprecated. Use files() instead.
+# Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy 
+# for migration advice.
+
 
 def get_environment_hook_template_path(name):
-    with importlib_resources.path('ament_package.template.environment_hook', name) as path:
-        return str(path)
+    # with importlib_resources.path('ament_package.template.environment_hook', name) as path:
+    path =  files('ament_package.template.environment_hook').joinpath(name)
+    print("WHATS FILES ", files('ament_package.template.environment_hook'))
+    path = str(path)
+    print("[PPP env hook template path] ", path, "\n")
+    return path
 
 
 def get_package_level_template_names(all_platforms=False):
@@ -41,8 +57,11 @@ def get_package_level_template_names(all_platforms=False):
 
 
 def get_package_level_template_path(name):
-    with importlib_resources.path('ament_package.template.package_level', name) as path:
-        return str(path)
+    # with importlib_resources.path('ament_package.template.package_level', name) as path:
+    path = files('ament_package.template.package_level').joinpath(name)
+    path = str(path)
+    print("[PPP package level template path] ", path, "\n")
+    return path
 
 
 def get_prefix_level_template_names(*, all_platforms=False):
@@ -61,8 +80,11 @@ def get_prefix_level_template_names(*, all_platforms=False):
 
 
 def get_prefix_level_template_path(name):
-    with importlib_resources.path('ament_package.template.prefix_level', name) as path:
-        return str(path)
+    # with importlib_resources.path('ament_package.template.prefix_level', name) as path:
+    path = files('ament_package.template.prefix_level').joinpath(name)
+    path = str(path)
+    print("[PPP prefix level template path] ", path, "\n")
+    return path
 
 
 def configure_file(template_file, environment):
