diff --git a/ament_package/templates.py b/ament_package/templates.py
index 8fda08d..497f0c8 100644
--- a/ament_package/templates.py
+++ b/ament_package/templates.py
@@ -22,10 +22,15 @@ except ModuleNotFoundError:
 
 IS_WINDOWS = os.name == 'nt'
 
+# importlib
+# DeprecationWarning: path is deprecated. Use files() instead.
+# Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy 
+# for migration advice.
 
 def get_environment_hook_template_path(name):
-    with importlib_resources.path('ament_package.template.environment_hook', name) as path:
-        return str(path)
+    # with importlib_resources.path('ament_package.template.environment_hook', name) as path:
+    path =  importlib_resources.files('ament_package.template.environment_hook').joinpath(name)
+    return str(path)
 
 
 def get_package_level_template_names(all_platforms=False):
@@ -41,8 +46,9 @@ def get_package_level_template_names(all_platforms=False):
 
 
 def get_package_level_template_path(name):
-    with importlib_resources.path('ament_package.template.package_level', name) as path:
-        return str(path)
+    # with importlib_resources.path('ament_package.template.package_level', name) as path:
+    path = importlib_resources.files('ament_package.template.package_level').joinpath(name)
+    return str(path)
 
 
 def get_prefix_level_template_names(*, all_platforms=False):
@@ -61,8 +67,9 @@ def get_prefix_level_template_names(*, all_platforms=False):
 
 
 def get_prefix_level_template_path(name):
-    with importlib_resources.path('ament_package.template.prefix_level', name) as path:
-        return str(path)
+    # with importlib_resources.path('ament_package.template.prefix_level', name) as path:
+    path = importlib_resources.files('ament_package.template.prefix_level').joinpath(name)
+    return str(path)
 
 
 def configure_file(template_file, environment):
